types/jquery-mousewheel/index.d.ts

```diff
@@ -2,22 +2,53 @@
 // Project: https://github.com/jquery/jquery-mousewheel
 // Definitions by: Brian Surowiec <https://github.com/xt0rted>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.3
+// Minimum TypeScript Version: 3.9
 
-/// <reference types="jquery"/>
+/// <reference types="jquery" />
 
-declare namespace JQueryMousewheel {
-    interface JQueryMousewheelEventObject extends JQueryEventObject {
-        deltaX: number;
-        deltaY: number;
-        deltaFactor: number;
-        deltaMode: number;
-        absDelta: number;
+declare global {
+    namespace JQueryMousewheel {
+        type JQueryMousewheelEventHook<
+            O = JQueryMousewheelEventObject,
+            D = JQueryMousewheelEventData
+        > = (
+            & { version: string; }
+            & { setup: (Extract<JQuery.SpecialEventHook<O, D>, { setup: any; }>)['setup'] }
+            & { teardown: (Extract<JQuery.SpecialEventHook<O, D>, { teardown: any; }>)['teardown'] }
+            & { getLineHeight(elem: Parameters<JQuery["appendTo"]>[0]): number | never; }
+            & { getPageHeight(elem: Parameters<JQuery["appendTo"]>[0]): number | never; }
+            & {
+                settings: {
+                    adjustOldDeltas: boolean;
+                    normalizeOffset: boolean;
+                };
+            }
+        );
+        type JQueryMousewheelEventHandler = (eventObject: JQueryMousewheelEventObject, ...args: any[]) => any;
+        interface JQueryMousewheelEventObject extends JQueryEventObject {
+            deltaX: number;
+            deltaY: number;
+            deltaFactor: number;
+            deltaMode: number;
+            absDelta: number;
+            offsetX: number;
+            offsetY: number;
+        }
+        interface JQueryMousewheelEventData {
+            'mousewheel-line-height': number;
+            'mousewheel-page-height': number;
+        }
+    }
+    namespace JQuery {
+        interface SpecialEventHooks {
+            mousewheel: JQueryMousewheel.JQueryMousewheelEventHook;
+        }
+    }
+    interface JQuery {
+        on(event: 'mousewheel', handler: JQueryMousewheel.JQueryMousewheelEventHandler): JQuery;
+        mousewheel(handler: JQueryMousewheel.JQueryMousewheelEventHandler): JQuery
+        unmousewheel(): JQuery;
     }
 }
 
-interface JQuery {
-    on(event: 'mousewheel', handler: (eventObject: JQueryMousewheel.JQueryMousewheelEventObject, ...args: any[]) => any): JQuery;
-    mousewheel(handler: (eventObject: JQueryMousewheel.JQueryMousewheelEventObject, ...args: any[]) => any): JQuery
-    unmousewheel(): JQuery;
-}
+export = undefined;
```

Comment:
from what I've gathered, the JS code for `jquery-mousewheel` is a plugin to `jquery`. And `jquery` can be used both as a global or imported as a commonjs module (i.e. using `require`), as you can see here: https://www.npmjs.com/package/jquery?activeTab=readme (section "Node").
So `jquery-mousewheel` also supports being used through `require`. From browsing the JS code, it looks like if you `require` it, it returns a factory function that modifies an object of type `JQueryStatic` (which `jquery` exports). So I think we should have something like:
```ts
declare function factory($: JQueryStatic): void;
export = factory;
```